File: remove-outermost-parantheses.js
function removeOuterParantheses(s) {
  let counter = 0;
  let ans = "";
  for (let i = 0; i < s.length; i++) {
    if (s[i] == "(") counter++;
    if (counter !== 0) ans += s[i];
    if (s[i] == ")") counter--;
  }
  return ans;
}

console.log(removeOuterParantheses("(((())))()())")); // out put: (((())))()()

File: remove-outermost-parantheses.py
class solution: 
    def removeOuterParantheses(self, s: str) -> str:
        counter = 0
        ans = ""

        for char in s:
            if char == ")":
                counter -= 1
            if counter != 0:
                ans += char
            if char == "(":
                counter += 1
        return ans

sol = solution()
print(sol.removeOuterParantheses("()((()()()((()))"))